src/v2/codegen/state.rs

```diff
@@ -372,19 +373,23 @@ pub mod util {
 #[cfg(feature = "cli")]
 impl EmitterState {
     /// Sets the crate metadata for this session.
-    pub fn set_meta(&mut self, meta: CrateMeta) {
-        self.crate_meta = Some(Rc::new(RefCell::new(meta)));
+    pub fn set_meta(&self, meta: CrateMeta) {
+        self.crate_meta = Rc::new(RefCell::new(Some(meta)));
```

Comment:
We should utilize refcell's interior mutability (`RefCell::borrow_mut`) here, right? :smile:

Reply:
Not sure if I got it right? 🤔  You mean `self.crate_meta.borrow_mut() = Some(meta);`

Reply:
`borrow_mut` produces a temporary borrow which can be dereferenced to the mutable version of meta. So,
```rust
*self.crate_meta.borrow_mut() = Some(meta);
```